src/sage/rings/function_field/drinfeld_modules/drinfeld_module.py

```diff
@@ -1061,17 +1272,117 @@ def is_finite(self):
         from sage.rings.function_field.drinfeld_modules.finite_drinfeld_module import FiniteDrinfeldModule
         return isinstance(self, FiniteDrinfeldModule)
 
-    def j_invariant(self):
+    def is_isomorphic(self, psi):
         r"""
-        Return the j-invariant of the Drinfeld module if the rank is
-        two; raise a NotImplementedError otherwise.
+        Return ``True``  whether ``self`` is isomorphic to the Drinfeld module
+        `\psi`.
 
-        Assume the rank is two. Write the generator `\phi_T = \omega +
-        g\tau + \Delta\tau^2`. The j-invariant is defined by
-        `\frac{g^{q+1}}{\Delta}`, `q` being the order of the base field
-        of the function ring. In our case, this field is always finite.
+        EXAMPLES::
 
-        OUTPUT: an element in the base codomain
+            sage: A = GF(5)['T']
+            sage: K.<T> = Frac(A)
+            sage: c = T^4 + 3*T^2 + T
+            sage: phi = DrinfeldModule(A, [T, T^3, T^9, T])
+            sage: psi = DrinfeldModule(A, [T, c^4*T^3, c^(24)*T^9, c^(124)*T])
+            sage: phi.is_isomorphic(psi)
+            True
+            sage: phi0 = DrinfeldModule(A, [T, 1, 1, 1])
+            sage: phi.is_isomorphic(phi0)
+            False
+
+        ::
+
+            sage: phi = DrinfeldModule(A, [T, 1, 0, 1, 0, 0, 0, T^2, 1, 1])
+            sage: phi.is_isomorphic(phi)
+            True
+            sage: psi = DrinfeldModule(A, [T, 1, 0, 1, 0, T, 0, T^2, 1, 1])
+            sage: phi.is_isomorphic(psi)
+            False
+        """
+        # Trivial checks:
+        if self._gen == psi._gen:
+            return True
+        if self._gen.degree() != psi._gen.degree():
+            return False
+        if self._gen.degree() == 1:
+            return True
+        # Check if self and psi are patterned alike:
+        if not all(bool(g_self) == bool(g_psi) for g_self, g_psi in\
+            zip(self._gen.coefficients(sparse=False),
+                psi._gen.coefficients(sparse=False))):
+            return False
+        # Check that all the nonzero basic j-invariants agree
+        isom = True
+        for p in self.basic_j_invariant_parameters(nonzero=True):
+            if self.j_invariant(p, check=False) != psi.j_invariant(p,check=False):
+                isom = False
+                break
+        return isom
```

Comment:
Here, there is one fix and one suggestion. 

1. The fix is a missing space before the second `check=False`.
2. The suggestion is the removal of `break`. This is shorter, and, I would argue, more elegant.

```suggestion
        for p in self.basic_j_invariant_parameters(nonzero=True):
            if self.j_invariant(p, check=False) != psi.j_invariant(p, check=False):
                return False
        return True
```

I would also add that `p` is not a great variable name. Although, in this situation, it allows for having the `if` on a single line, and we can keep it that way.